packages/apify/src/actor.ts

```diff
@@ -1545,11 +1587,94 @@ export class Actor<Data extends Dictionary = Dictionary> {
      * @ignore
      */
     newClient(options: ApifyClientOptions = {}): ApifyClient {
+        const actor = this;
+
         const { storageDir, ...storageClientOptions } = this.config.get(
             'storageClientOptions',
         ) as Dictionary;
         const { apifyVersion, crawleeVersion } = getSystemInfo();
-        return new ApifyClient({
+
+        class PatchedDatasetClient<
+            Data extends Record<string | number, any> = Record<
+                string | number,
+                unknown
+            >,
+        > extends DatasetClient<Data> {
+            override async pushItems(
+                items: string | Data | string[] | Data[],
+            ): Promise<void> {
+                const context = actor.ppeAwarePushDataContext.getStore();
+                const eventName =
+                    context?.eventName ??
+                    (this.id === actor.config.get('defaultDatasetId')
+                        ? 'apify-default-dataset-item'
+                        : undefined);
+
+                if (eventName === undefined) {
+                    // No charging needed
+                    await super.pushItems(items);
+                    return;
+                }
+
+                const { limitedItems, countToCharge } =
+                    actor.calculatePushDataLimits(items, eventName);
+
+                if (countToCharge > 0) {
+                    await super.pushItems(limitedItems as string[] | Data[]);
+                }
+
+                const chargeResult = await actor.chargingManager.charge({
+                    eventName,
+                    count: countToCharge,
+                });
```

Comment:
We shouldn't be charging platform via API here. So this might need either direct increment or new method.